WinUI3 - Settings page - Accounts

herve-er opened the pull request:
https://github.com/user-attachments/assets/4f81b722-913d-4f29-8231-af16cff2701d


The current wording of the disconnect dialog is known to be inaccurate; the UX team is working on an updated version.

herve-er committed:
feat(server-communication): add support for availables drives

herve-er committed:
feat(settings): improve drive management and UI templates

Added new DataTemplates for configured and unconfigured drives with a DriveDataTemplateSelector. Introduced the IDrive interface to unify drive properties. Updated the User class with a combined AllDrives collection and added methods to manage available drives. Refactored copy methods and improved nullable property handling. Simplified drive management logic in ServerCommService.

herve-er committed:
feat(settings): enhance UI and add user management features

- Added `SettingsExpander` resource in `App.xaml` for new UI components.
- Implemented event handlers for "Manage" and "Activate" buttons in `SettingsPage.xaml`.
- Updated `ListViewItem` styles to use a static resource for margins.
- Added a footer in `SettingsExpander` to allow user account disconnection.
- Introduced `DisconectUser_Click` method with a confirmation dialog.
- Optimized drive management in `ServerCommService.cs` to reduce UI flickering.
- Added `DriveId` property to `IDrive` interface for unique identification.
- Refactored `MergeDrives` in `User.cs` to avoid duplicates.
- Cleaned up unused imports across multiple files.
- Added debug logs for better traceability.

herve-er committed:
feat(settings): Add localization of the account section and loading ring

Added `ProgressRing` to indicate loading state in user settings.
Updated dialogs to use localized strings.

ClementKunz commented on src/gui4/windows/kDrive client/kDrive client/ServerCommunication/Services/ServerCommService.cs:
```diff
@@ -213,6 +214,65 @@ await Utility.RunOnUIThread(() =>
+                ["userDbId"] = userDbId
+            };
+            CommData data = await _commClient.SendRequestAsync(RequestNum.USER_AVAILABLEDRIVES, parms, cancellationToken);
+            if (data.Params == null || !data.Params.ContainsKey("driveAvailableInfoList"))
```
Should we define those strings (i.e. `userDbId`, `driveAvailableInfoList`, ...) in static variable (or anything else) somewhere to avoid hardcoding them everywhere?

herve-er replied:
Done here: [refactor(comm WinUI): centralize JSON keys with JsonKeys](https://github.com/Infomaniak/desktop-kDrive/pull/1161/commits/e633b8629bf220777197999bb8aff8d7340767c4)

herve-er committed:
refactor(localization): migrate to Utility.GetLocalizedString

Replaced ResourceLoader calls with Utility.GetLocalizedString
across the application to standardize localization handling.

- Updated converters to use Utility.GetLocalizedString.
- Refactored custom controls and pages for consistent localization.
- Renamed and reorganized resource keys in Resources.resw.
- Removed unused dependencies in multiple files.
- Added new DataTemplates in SettingsPage.xaml.
- Improved dialogs and user-facing messages for better UX.

luc-guyot-infomaniak commented on src/gui4/windows/kDrive client/kDrive client/ViewModels/App/User.cs:
```diff
@@ -137,5 +154,39 @@ public ReadOnlyObservableCollection<Drive> Drives
+                if (drives.FirstOrDefault(d => d.DriveId == drive.DriveId) is null)
+                    drives.Add(drive);
+
+            for (int i = AllDrives.Count - 1; i >= 0; i--)
```
It would be nice to explain briefly what is being done. It looks as if `drives` and `AllDrives` are beeing "equalized".

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1161/commits/13e37a2abb61befae12aaced540f2674e3b252da

luc-guyot-infomaniak replied:
Is the commit https://github.com/Infomaniak/desktop-kDrive/commit/13e37a2abb61befae12aaced540f2674e3b252da the correct one? I see no change on this file.

herve-er replied:
My bad sorry, here is the correct one: https://github.com/Infomaniak/desktop-kDrive/pull/1161/commits/09c03d83b5a9a82bf88ff2cb65ac4c48d6befa9e

herve-er committed:
feat(user-management): add user removal functionality

Introduce `RemoveUser` method in `IServerCommService` and its
implementation in `ServerCommService`. Handle `USER_REMOVED`
signal to update the ViewModel. Temporarily disable controls
during user disconnection in `SettingsPage.xaml.cs`.

herve-er committed:
refactor(comm WinUI): centralize JSON keys with JsonKeys

Introduce a centralized JsonKeys structure for managing JSON keys.
Replace hardcoded strings with constants across the codebase.
Update logging, serialization, and key checks for consistency.

herve-er committed:
chore(viewmodels): add documentation on drive merging logic

herve-er committed:
fix(SettingsPage): handle null item in SelectTemplateCore

Add a null check for the `item` parameter in the `SelectTemplateCore`
method to ensure the default behavior is used when `item` is null.
This prevents potential null reference exceptions.

herve-er committed:
feat(settings): add sync rules settings card

Added a new settings card in `SettingsPage.xaml` for configuring sync rules. The card includes a description, an icon, and a click event handler.

Added the `SyncRulesCard_Clicked` method in `SettingsPage.xaml.cs` to handle navigation to the sync rules page (TODO implementation pending).

Updated existing methods with TODO comments for navigation to the appropriate pages.

ClementKunz committed:
Merge pull request #1164 from Infomaniak/WinUI3-Settings-page-sync-rules

WinUI3 settings page sync rules
